x/evm/keeper/grpc_query.go

```diff
@@ -499,17 +512,27 @@ func (k Keeper) TraceBlock(c context.Context, req *types.QueryTraceBlockRequest)
 	if err != nil {
 		return nil, status.Error(codes.Internal, "failed to load evm config")
 	}
-	signer := ethtypes.MakeSigner(cfg.ChainConfig, big.NewInt(ctx.BlockHeight()))
+	height := ctx.BlockHeight()
+	patch := height < req.FixClearAccessListHeight
+	signer := ethtypes.MakeSigner(cfg.ChainConfig, big.NewInt(height))
 	txsLength := len(req.Txs)
 	results := make([]*types.TxTraceResult, 0, txsLength)
-
 	txConfig := statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash().Bytes()))
+	var lastDB *statedb.StateDB
 	for i, tx := range req.Txs {
 		result := types.TxTraceResult{}
 		ethTx := tx.AsTransaction()
 		txConfig.TxHash = ethTx.Hash()
 		txConfig.TxIndex = uint(i)
-		traceResult, logIndex, err := k.traceTx(ctx, cfg, txConfig, signer, ethTx, req.TraceConfig, true, nil)
+		var stateDB *statedb.StateDB
+		if patch {
+			stateDB = statedb.New(ctx, &k, txConfig)
+			if lastDB != nil {
+				stateDB.SetAddressToAccessList(lastDB.GetAddressToAccessList())
+			}
+			lastDB = stateDB
```

Comment:
Is it possible to add tests for this functionality? Testing your getter and setter methods (`GetAddressToAccessList`, `SetAddressToAccessList`) would be covered by tests for this section as well.

Reply:
Added